src/components/lab/EmittedLine.js

```diff
@@ -33,13 +33,22 @@ class EmittedLine extends Component {
     t: 0,
   };
 
-  componentDidUpdate({ isPaused: prevIsPaused }) {
-    const { isPaused } = this.props;
+  componentDidUpdate({
+    isPaused: prevIsPaused,
+    showEmittedLines: prevShowEmittedLines,
+  }) {
+    const { isPaused, showEmittedLines } = this.props;
     if (isPaused !== prevIsPaused && isPaused) {
       clearInterval(this.emittedLineInterval);
     } else if (isPaused !== prevIsPaused && !isPaused) {
       this.beginLineInterval();
     }
+
+    // reset line on show/hide emitted lines
+    if (prevShowEmittedLines !== showEmittedLines) {
```

Comment:
I remember discussing a similar situation with @juancarlosfarah for application 1, and he recommended that to be extra sure not to cause an infinite loop, to add to the `if` condition:

``&& !_.isEqual(prevState.points, [0, 0])``